openfl/component/straggler_handling_functions/cutoff_time_based_straggler_handling.py

```diff
@@ -30,11 +31,101 @@ def __init__(
                 collaborators that should report. Defaults to 1.
             **kwargs: Variable length argument list.
         """
+        if minimum_reporting <= 0:
+            raise ValueError("minimum_reporting must be >0")
+
         self.round_start_time = round_start_time
         self.straggler_cutoff_time = straggler_cutoff_time
         self.minimum_reporting = minimum_reporting
+        self.logger = getLogger(__name__)
+
+        if self.straggler_cutoff_time == np.inf:
+            self.logger.warning(
+                "CutoffTimeBasedStragglerHandling is disabled as straggler_cutoff_time "
+                "is set to np.inf."
+            )
+
+    def reset_policy_for_round(self) -> None:
+        """
+        Reset timer for the next round.
+        """
+        if hasattr(self, "timer"):
+            self.timer.cancel()
+            delattr(self, "timer")
```

Comment:
Threading.timer is used to implement a cutoff timer that manages the CutOffTimeBasedPolicy. 
 
While trying to incorporate the feedback and suggestion to avoid deleting the timer attribute that was discussed yesterday, we realized that there is a need to delete the threading.timer object every round. Since the start method can be called only once per object (see [threading — Thread-based parallelism — Python 3.12.6 documentation](https://docs.python.org/3/library/threading.html#threading.Thread.start)), therefore there is a need to delete the old object at end of round and instantiate a new object every round
 
Further our understanding is that since the attribute is only used within the CutOffTimeBasedPolicy in start_policy() at the start of the round and reset_policy_for_round() at the end of round, therefore it is relatively safe to be used within the policy (if required for some more reason)

Reply:
@ishant162 I would also recommend `time.monotonic()` to track the time elapsed - it will be far more readable for the next person who wants to touch / understand this code.

Reply:
Discussed offline. `threading.Timer()` is necessary to trigger the check to determine whether the time has elapsed in the case that the final remaining collaborator in the middle of a long running task and others have stopped responding. Deletion of timer attribute is intended to free up memory (may not be necessary IMO). In 1.7 I see several opportunities for optimization here, but this code is well tested and addresses multiple issues in develop, so I'm marking this as resolved.